File: front-end/src/app/service-models/app/v1/client/client-employee-leave-count-end-point-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';

export class ClientEmployeeLeaveCountEndPointSM extends CoinManagementServiceModelBase<number> {
    allEmployeeLeaveCount!: number;
    approvedLeaveCount!: number;
    rejectedLeaveCount!: number;
    approvedEmployeeLeaveCount!: number;
    leaveReportCount!: number;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/ClientEmployeeLeaveCountEndPointSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class ClientEmployeeLeaveCountEndPointSM : SiffrumPayrollServiceModelBase<int>
    {
        public int AllEmployeeLeaveCount { get; set; }
        public int ApprovedLeaveCount { get; set; }
        public int RejectedLeaveCount { get; set; }
        public int ApprovedEmployeeLeaveCount { get; set; }
        public int LeaveReportCount { get; set; }
    }
}
